DashKit/DashPumpManagerState.swift

```diff
@@ -213,6 +193,19 @@ public struct DashPumpManagerState: RawRepresentable, Equatable {
         podActivatedAt = status.expirationDate - .days(3)
         podTotalDelivery = status.delivered
     }
+    
+    mutating func finalizeDoses() {
+        if let bolus = unfinalizedBolus, bolus.isFinished(at: dateGenerator()) {
```

Comment:
Is there a unit test that checks both of these (or rather all 4 of these) code paths?